Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/task/edit/presenter/EditTaskPresenter.java

```diff
@@ -0,0 +1,57 @@
+package com.novoda.todoapp.task.edit.presenter;
+
+import com.novoda.data.SyncedData;
+import com.novoda.event.DataObserver;
+import com.novoda.todoapp.navigation.Navigator;
+import com.novoda.todoapp.task.TaskActionListener;
+import com.novoda.todoapp.task.data.model.Task;
+import com.novoda.todoapp.task.edit.displayer.EditTaskDisplayer;
+import com.novoda.todoapp.task.presenter.IdProducer;
+import com.novoda.todoapp.tasks.service.TasksService;
+
+import rx.subscriptions.CompositeSubscription;
+
+public class EditTaskPresenter {
+
+    private final TasksService tasksService;
+    private final EditTaskDisplayer taskDisplayer;
+    private final IdProducer idProducer;
+
+    final TaskActionListener taskActionListener;
```

Comment:
Can do, maybe in another PR since this means testing the `taskActionListener` as independant collaborator. It might add a bit too much noise to this PR.